src/Microsoft.AspNetCore.SystemWebAdapters/Caching/CacheDependency.cs

```diff
@@ -135,14 +137,27 @@ protected virtual void Dispose(bool disposing)
         {
             if (disposing)
             {
-                foreach (var changeMonitor in changeMonitors)
+                if (!disposedValue)
```

Comment:
NIT: Do we actually need the three checks for `!disposedValue` or could we get away with just two by reordering the first check? Just mentioning it since typical locks usually only do two checks.

Reply:
Yup - that's what I meant to do. I'll fix that up